defaults/main.yml

```diff
@@ -1,5 +1,5 @@
 ---
-mysql_user_home: /root
+mysql_user_home: "~{{ ansible_ssh_user }}"
```

Comment:
That's actually what I initially tried.  And it works in almost every case, since it's the same thing.  The problem I ran into was, e.g. running the playbook as a `vagrant` user, but the `ansible` user initially bootstrapped the mysql installation, so only `ansible` has the credentials file.  This hopefully accounts for the situation where `ansible_ssh_user` has been explicitly set in `ansible.cfg` (which was my situation).